### PR TITLE
Update grdfilter usage

### PR DESCRIPTION
**Pull request by PaulWessel**
Follows the recipe of #5341.

### CODE REVIEW

File: src/grdfilter.c [hunk 1]
```diff
@@ -584,72 +584,79 @@ GMT_LOCAL struct GMT_GRID *init_area_weights (struct GMT_CTRL *GMT, struct GMT_G
...
-	GMT_Message (API, GMT_TIME_NONE, "\t        Append +u to return the uppermost mode if multiple modes are found [return average].\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t     u: Upper : return maximum of all points.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t     U: Upper- : return maximum of all negative points.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t     If <width> is a grid it must match output domain and registration\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-G Set output filename for filtered grid.\n");
+	GMT_Usage (API, 4, "+l Return the lowest mode if multiple modes are found [return average].");
+	GMT_Usage (API, 4, "+u Return the uppermost mode if multiple modes are found [return average].");
+	GMT_Usage (API, 3, "u: Upper : return maximum of all points.");
+	GMT_Usage (API, 3, "U: Upper- : return maximum of all negative points.");
+	GMT_Usage (API, -2, "If <width> is a grid it must match the output domain and registration.");
```
Comment (KristofKoch) [inline src/grdfilter.c:637]: Which width is meant by `<width>`? `<filter_width>[/<width2>]`

File: src/grdfilter.c [hunk 2]
```diff
@@ -584,72 +584,79 @@ GMT_LOCAL struct GMT_GRID *init_area_weights (struct GMT_CTRL *GMT, struct GMT_G
...
+		"Next three options are slower; weights must be recomputed for each scan line.:");
+	GMT_Usage (API, 3, "3: grid x,y in degrees, <filter_width> in km, x scale varies as cos(y), Cartesian distances.");
+	GMT_Usage (API, 3, "4: grid x,y in degrees, <filter_width> in km, spherical distances.");
+	GMT_Usage (API, 3, "5: grid x,y in Mercator units (-Jm1), <filter_width> in km, spherical distances.");
+	GMT_Usage (API, 1, "\n-F<type><filter_width>[/<width2>][<modifiers>]");
+	GMT_Usage (API, -2, "Set the low-pass filter type and full diameter (6 sigma) filter-width. "
+		"Choose between convolution-type filters which differ in how weights are assigned "
+		"and geospatial filters that seek to return a representative value. "
+		"Append +h to select high-pass filtering [Default is low-pass filtering]. "
+		"Filters are isotropic.  For rectangular filtering append /<width2> (requires -Dp|0). "
```
Comment (KristofKoch) [inline src/grdfilter.c:611]:
```suggestion
		"Filters are isotropic.  For rectangular filtering append /<width2> (requires -Dp|0).");
```
New line for nicer layout (1/2)

File: src/grdfilter.c [hunk 3]
```diff
@@ -584,72 +584,79 @@ GMT_LOCAL struct GMT_GRID *init_area_weights (struct GMT_CTRL *GMT, struct GMT_G
...
+		"Convolution filters:");
```
Comment (KristofKoch) [inline src/grdfilter.c:612]:
```suggestion
	GMT_Usage (API, -2, "Convolution filters:");
```
New line for nicer layout (2/2)

File: src/grdfilter.c [hunk 4]
```diff
@@ -584,72 +584,79 @@ GMT_LOCAL struct GMT_GRID *init_area_weights (struct GMT_CTRL *GMT, struct GMT_G
...
-	GMT_Message (API, GMT_TIME_NONE, "\t-A DEBUG: Use -A<mode><lon/<lat> to instead save filter specifics at that point. Choose:\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   mode as a (area weights), c (composite weight), r (radii), or w (filter weight).\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-A DEBUG: Save area weights to <file> with -As<file> [no save]\n");
+	GMT_Usage (API, 1, "\n-A<mode><lon/<lat>");
+	GMT_Usage (API, -2, "DEBUG: Use -A<mode><lon/<lat> to instead save filter specifics at that point. Choose: "
+		"mode as a (area weights), c (composite weight), r (radii), or w (filter weight).");
+	GMT_Usage (API, 1, "\n-As<file>");
+	GMT_Usage (API, -2, "DEBUG: Save area weights to <file> with -As<file> [no save]");
 #endif
 	GMT_Option (API, "I");
```
Comment (KristofKoch) [inline src/grdfilter.c:648]: `+e` and `+n` could need one more indentation level.

I-option

Comment (PaulWessel) [reply src/grdfilter.c:648]: Yes, but already fixed in another PR so not doing it it this one.  All will be well when both are merged.

File: src/grdfilter.c (hunk 1 above)
Comment (PaulWessel) [reply src/grdfilter.c:637]: Revised to use <width> and <width2> instead of the unnecessary filter_ prefix.

File: src/grdfilter.c (hunk 2 above)
Comment (PaulWessel) [reply src/grdfilter.c:611]: OK, done

File: src/grdfilter.c (hunk 3 above)
Comment (PaulWessel) [reply src/grdfilter.c:612]: DOne

**Comment by PaulWessel** [comment]
Should be OK now?

**Review by KristofKoch — APPROVED** [review]
Looks good!

**Pull request merged**